File: JavaScript/1-parent.js
'use strict';

function Node(parent, data) {
  this.parent = parent;
  this.data = data;
}

// Usage

const root = new Node(null, { name: 'root' });
const n1 = new Node(null, { name: 'n1' });

const n2 = new Node(n1, { name: 'n2' });
const n3 = new Node(n1, { name: 'n3' });
const n4 = new Node(n1, { name: 'n4' });

const n21 = new Node(n2, { name: 'n21' });
const n22 = new Node(n2, { name: 'n22' });

const n31 = new Node(n3, { name: 'n31' });
const n32 = new Node(n3, { name: 'n32' });
const n33 = new Node(n3, { name: 'n33' });
const n34 = new Node(n3, { name: 'n34' });

console.dir(n34);
console.dir(n34.parent);
console.dir(n34.parent.parent);
console.dir(n34.parent.parent.parent);

File: Java/src/1-parent.java
class Parent {
    public static void main(String[] args) {
        Node root = new Node(null, "root");
        Node n1 = new Node(null, "n1");

        Node n2 = new Node(n1, "n2");
        Node n3 = new Node(n1, "n3");
        Node n4 = new Node(n1, "n4");

        Node n21 = new Node(n2, "n21");
        Node n22 = new Node(n2, "n22");

        Node n31 = new Node(n3, "n31");
        Node n32 = new Node(n3, "n32");
        Node n33 = new Node(n3, "n33");
        Node n34 = new Node(n3, "n34");

        System.out.println(n34);
        System.out.println(n34.parent);
        System.out.println(n34.parent.parent);
        System.out.println(n34.parent.parent.parent);
    }
    static class Node {
        public Node parent;
        public Object data;

        public Node(Node parent, Object data) {
            this.parent = parent;
            this.data = data;
        }

        @Override
        public String toString() {
            return "Node {" +
                    "parent=" + parent +
                    ", data=" + data +
                    '}';
        }
    }
}
